data/core/docview.lua

```diff
@@ -97,7 +97,16 @@ end
 
 
 function DocView:get_scrollable_size()
-  return self:get_line_height() * (#self.doc.lines - 1) + self.size.y
+  local xmargin = 3 * self:get_font():get_width(' ') -- from DocView:scroll_to_make_visible
+  local long_line = 1
+  for l,_ in pairs(self.doc.long_lines.line_numbers) do -- get any of the longest lines
+    long_line = l
+    break
+  end
+  local size_v = self:get_line_height() * (#self.doc.lines - 1) + self.size.y
+  local size_h = self:get_col_x_offset(long_line, self.doc.long_lines.length)
+                 + self.size.x - (self.size.x - self:get_gutter_width()) + xmargin
```

Comment:
+ self.size.x - self.size.x ?

Reply:
It was to keep a similar style of `+ self.size.y` that then gets subtracted in `clamp_scroll_position`. But yeah, it looks redundant, I'll remove it.